process/block/baseProcess.go

```diff
@@ -1294,19 +1294,55 @@ func (bp *baseProcessor) commitTrieEpochRootHashIfNeeded(metaBlock *block.MetaBl
 	}
 
 	balanceSum := big.NewInt(0)
+	nrAccounts := 0
+	nrAccountsWithDataTrie := 0
+	nrNotAccounts := 0
+	totalSizeAccounts := 0
+	totalSizeAccountsDataTries := 0
+	totalSizeNonAccounts := 0
 	for leaf := range allLeavesChan {
 		userAccount, errUnmarshal := unmarshalUserAccount(leaf.Key(), leaf.Value(), bp.marshalizer)
 		if errUnmarshal != nil {
+			nrNotAccounts++
+			totalSizeNonAccounts += len(leaf.Value())
 			log.Trace("cannot unmarshal user account. it may be a code leaf", "error", errUnmarshal)
 			continue
 		}
+
+		rh := userAccount.GetRootHash()
```

Comment:
These operations can be costly for a large state. You should do them only if the log level is set to debug.

Reply:
fixed. Added another debug flag